explainer-datamodel.md

```diff
@@ -1,236 +1,331 @@
-from inline to block (taking it out-of-flow). `position: absolute` also takes
-the object out of the normal flow. Likewise, the `width` and `height` properties would
+from inline to block (taking it out-of-flow). `position: absolute` as a metadata property also takes
+the object out of the normal flow. Likewise, the `width` and `height` properties on specific text runs would
 be ignored because they do not apply to inline-level elements. Attempting to set these 
-properties would have no effect. On the other hand, `padding` would be supported and 
-applied to `FormattedTextRun` objects in accordance with the rules of CSS inline layout,
+properties would have no effect. On the other hand, `padding` **would be supported** and 
+applied to text runs in accordance with the rules of CSS inline layout,
 e.g., the inline-direction `padding` values are taken into account while the 
 block-direction values are not.
 
-The `FormattedText` object is an independent containing block for text. It is essentially a 
+The container for text runs is an independent containing block. It is essentially a 
 `display: flow-root` object that establishes an inline formatting context for its children.
-The `FormattedText` object has no positional relationship with anything other than its
-children, and so values like `top`, `right`, `bottom`, and `left` do not apply (i.e., the 
-object acts as its own initial containing block). Positioning the object for the purpose of
-rendering in some context must be done separately. In order to render or extract useful metrics
-from the `FormattedText` object, final dimensions for its size must be supplied (see 
-[Rendering explainer](explainer-rendering.md)) which are used to resolve relative values in
-supported properties like `width` and `height`.
-
-In some cases, we imagine it will be useful to allow the `FormattedText` object to assume
-alternate layout container types where those alternate types provide unique text layout
-capabilities. For example, we expect to support `FormattedText` objects with an inner 
-display type of `ruby` in order to become a Ruby container and enable the use of Ruby
-annotated layout. Other container types are not currently planned to initially support 
+When used in the metadata parameter, properties like `top`, `right`, `bottom`, and `left` do
+not apply (i.e., the text run container acts as its own initial containing block). Positioning
+the resulting formatted text for the purpose of rendering in some context must be done separately.
+
+In some cases, we imagine it will be useful to allow `format` metadata properties to support 
+some limited alternate layout container types where those alternate types provide unique text layout
+capabilities. For example, we expect to support an inner display type of `ruby` in order to 
+become a Ruby container and enable the use of Ruby annotated layout. Other container types are
```

Comment:
This is a can of worms.  I guess it would be relatively easy to support a single top-level Ruby block, but what about Ruby inside a span?  E.g., how would you format something like this?  `<div>Some <ruby>text<rt>annotated</rt></ruby>.</div>`

Reply:
I have a lot to learn about ruby, so I'm going to 🤷‍♂️ and hope we can figure it out!